mr/src/main/java/org/apache/iceberg/mr/mapreduce/IcebergInputFormat.java

```diff
@@ -166,14 +168,32 @@ private static void checkResiduals(CombinedScanTask task) {
   }
 
   private static final class IcebergRecordReader<T> extends RecordReader<Void, T> {
+
+    private static final String HIVE_VECTORIZED_READER_CLASS = "org.apache.iceberg.mr.hive.vector.HiveVectorizedReader";
+    private static final DynMethods.StaticMethod HIVE_VECTORIZED_READER_BUILDER;
+
+    static {
+      if (MetastoreUtil.hive3PresentOnClasspath()) {
+        HIVE_VECTORIZED_READER_BUILDER = DynMethods.builder("reader")
```

Comment:
Are these classes are missing in Hive2?

Reply:
This is the HiveVectorizedReader utility class that belongs to Iceberg codebase, it is located in iceberg-hive3 module. Vectorization for Hive is only supported in iceberg-hive3 module, so here in the MR module we need to dynamically load this class (IcebergInputFormat is reused in iceberg-hive3 too)